src/js/d3ViewComponent.js

```diff
@@ -114,11 +139,101 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
     };
 
     // Returns a formatted string for a numeric data value based on a supplied template
-    floe.d3ViewComponent.getTemplatedDisplayValue = function(totalValue, percentageDigits, template, d) {
+    floe.d3ViewComponent.getTemplatedDisplayValue = function (totalValue, percentageDigits, template, d) {
         var percentage = floe.chartAuthoring.percentage.calculate(d.value, totalValue);
         var percentageForTemplate = percentage !== null ? percentage.toFixed(percentageDigits) : percentage;
         var output = fluid.stringTemplate(template, {label: d.label, value: d.value, percentage: percentageForTemplate, total: totalValue});
         return output;
     };
 
+    // Given a data key used to maintain object constancy in D3, a DOM
+    // element with a unique ID and the component, updates the key's
+    // value (an array of IDs) to include that ID
+    floe.d3ViewComponent.addElementIdToDataKey = function (d3Key, idToAdd, that) {
+        var keyPath = "dataKeys." + d3Key;
```

Comment:
We have all been weaning ourselves off the same unhealthy preference in a very long process.... as the last 7 years of framework development shows, where so many of our original configuration elements were formulated as arrays (distributeOptions, modelListeners, etc.). Part of array-ism is the "save the nanoseconds" movement, but the other part of it is "not planning for design extension" - arrays are very hard to share amongst multiple creators since their indices can't be depended on to be stable.